src/io/GeantImporter.cc

```diff
@@ -0,0 +1,152 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file GeantImporter.cc
+//---------------------------------------------------------------------------//
+
+#include <iomanip>
+#include <iostream>
+
+// ROOT
+#include <TFile.h>
+#include <TTree.h>
+#include <TBranch.h>
+#include <TLeaf.h>
+
+#include "GeantImporter.hh"
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Constructor and destructor
+ */
+GeantImporter::GeantImporter(std::string filename)
+{
+    this->root_input_.reset(TFile::Open(filename.c_str(), "read"));
+}
+
+GeantImporter::~GeantImporter() = default;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Operators ()
+ */
+geant_importer_data GeantImporter::operator()()
+{
+    geant_importer_data geant_data;
+    load_particle_data();
+    geant_data.particle_params = get_ParticleParams();
+
+    // Placeholder
+    // load_physics_tables();
+    // geant_data.physics_tables = get_PhysicsTables();
+
+    return geant_data;
+}
+
+geant_importer_data
+GeantImporter::operator()(std::vector<PDGNumber> pdg_numbers)
+{
+    geant_importer_data              geant_data;
+    ParticleParams::VecAnnotatedDefs defs;
+    load_particle_data();
+
+    for (PDGNumber pdg : pdg_numbers)
+    {
+        for (GeantParticle particle : this->particle_vector_)
+        {
```

Comment:
Since the ROOT file is likely to have more data than the subset of PDG numbers being passed, it's better to have that on the outer loop if the inner loop is doing a linear search. But regardless, this function is again doing a possibly N^2 loop over particles because it's doing linear searches. It also doesn't have any assertions about the user-requested PDG numbers being there. It also doesn't save any time or memory loading the data since it simply truncates the data after it's read.

When I was suggesting this function, it was sort of a hypothetical "this might be a useful way to overload `operator()`". Since we don't need this immediately, how about for the time being we remove it until we need it?